src/caches/low_storage_rk_caches.jl

```diff
@@ -0,0 +1,366 @@
+
+@cache struct CarpenterKennedy2N54Cache{uType,rateType,TabType} <: OrdinaryDiffEqMutableCache
+  u::uType
+  uprev::uType
+  k::rateType
+  tmp::uType
+  fsalfirst::rateType
+  tab::TabType
+end
+
+struct CarpenterKennedy2N54ConstantCache{T,T2} <: OrdinaryDiffEqConstantCache
+  A2end::SVector{4,T} # A1 is always zero
+  B1::T
+  B2end::SVector{4,T}
+  c2end::SVector{4,T2} # c1 is always zero
+
+  function CarpenterKennedy2N54ConstantCache(::Type{T}, ::Type{T2}) where {T,T2}
+    A2 = T(-567301805773/1357537059087)
```

Comment:
not all number types accept this constructor, so do `convert(T,...)`. Also, it's good to make it compatible with arbitrary precision, which just amounts to making the coefficients rational `//` since otherwise the coefficients are already chopped.

Reply:
I will do that. But the rational coefficients aren't really rational, i.e. the rational forms are also only approximations and no exact solutions of the order conditions.